src/Internal/Payment/State/InitialState.php

```diff
@@ -194,6 +199,28 @@ protected function populate_context_from_order() {
 		$context->set_customer_id( $customer_id );
 	}
 
+	/**
+	 * Validates the order phone number, and run necessary actions if invalid.
+	 *
+	 * @return PaymentErrorState|null    The error state if the phone number is valid. Null otherwise.
+	 * @throws Order_Not_Found_Exception
+	 * @throws StateTransitionException
+	 * @throws ContainerException
+	 */
+	protected function process_order_phone_number(): ?PaymentErrorState {
+		$context  = $this->get_context();
+		$order_id = $context->get_order_id();
+
+		if ( ! $this->order_service->is_valid_phone_number( $order_id ) ) {
+			$context->set_error_message(
```

Comment:
The exception handling for intent requests in `start_processing` was just an excuse to use the state, and show it can be generated, but it will probably not remain this way, and is a bad example. Right now, it does not make much sense to add all the additional logic to transition to a new payment state, store the message, etc, because those would not be persisted anyway. An exception, thrown here, can simply bubble up to the gateway without much logic required, and would achieve the exact same result. Error handling additions will be managed as a part of a future stream.

For now, let's throw an exception here immediately, and remove the `if()` from `start_processing`.

Reply:
Agreed, do it here 4c635e0a3437349b2ac8bc03bd4e9ffa069431df